experimental/authclient/authclient.go

```diff
@@ -0,0 +1,42 @@
+package authclient
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+)
+
+// New auth client which is basically a http client but specific functionalities implemented depends on AuthMethod
+func New(httpOptions httpclient.Options, options AuthOptions) (client *http.Client, err error) {
+	defaultClient, err := httpclient.New(httpOptions)
+	if err != nil {
+		return nil, err
+	}
+	switch options.AuthMethod {
+	case AuthMethodOAuth2:
```

Comment:
This is what I think doesn't make sense from the code perspective. If you name this package oauth2client, you don't need the `AuthMethod`. The client, rather than setting this Method based on the user input, can call the normal `httpClient.New` or this `oauth2Client.New`.

If in the future, there is a third auth type, then you can create the abstraction of an `authclient` that can either instantiate an `httpClient`, an `oauth2Client` or something else but for the case of only 2 types I don't think it's needed this extra complexity here.

Reply:
I am not a fan of merging code that is meant for the future but since this is still `experimental` it's not a blocker :+1: